infra/blueprint-test/pkg/gcloud/gcloud.go

```diff
@@ -20,24 +20,23 @@ package gcloud
 import (
 	"strings"
 
-	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/utils"
-	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/binary"
 	"github.com/gruntwork-io/terratest/modules/shell"
 	"github.com/mitchellh/go-testing-interface"
 	"github.com/tidwall/gjson"
 )
 
 type CmdCfg struct {
-	gcloudBinary string         // path to gcloud binary
-	commonArgs   []string       // common arguments to pass to gcloud calls
-	logger       *logger.Logger // custom logger
+	*binary.BinaryCfg                       // binary config
+	bOpts             []binary.BinaryOption // binary options
+	commonArgs        []string              // common arguments to pass to gcloud calls
 }
 
 type cmdOption func(*CmdCfg)
 
-func WithBinary(gcloudBinary string) cmdOption {
+func WithBinaryOptions(bOpts ...binary.BinaryOption) cmdOption {
```

Comment:
We shouldn't need this layer of indirection. As a consumer of the `gcloud` package, I shouldn't need to know if options are common (across all binaries) or specific to gcloud.

Reply:
@morgante in that case should we continue to keep the `WithBinary` and other options within the `gcloud` pkg?

Reply:
@bharathkkb No, I think it should still be possible to share the options by using a interfaces. If this doesn't work I can give it a try.

Reply:
Thanks, I'll try that approach